### PR TITLE
Scale - wait for resource is getting a list of nested PVC list instead of PVC obj list. 

### PR DESCRIPTION
**Pull request by shyRozen**
Adding the option to handle nested lists in the creation of PVCs.

Fixes: #6268 
Signed-off-by: Shay Rozen <shay.rozen@gmail.com>

### CODE REVIEW

File: ocs_ci/helpers/helpers.py [hunk 1]
```diff
@@ -2162,9 +2162,15 @@ def create_multiple_pvc_parallel(sc_obj, namespace, number_of_pvc, size, access_
     # Check for all the pvcs in Bound state
     with ThreadPoolExecutor() as executor:
         for objs in pvc_objs_list:
-            obj_status_list.append(
-                executor.submit(wait_for_resource_state, objs, "Bound", 90)
-            )
+            if type(objs) is list:
+                for obj in objs:
+                    obj_status_list.append(
+                        executor.submit(wait_for_resource_state, obj, "Bound", 90)
```
Comment (ebenahar) [inline ocs_ci/helpers/helpers.py:2168]: would status always be `Bound` for any instance?

Comment (shyRozen) [reply ocs_ci/helpers/helpers.py:2168]: Why any instance. This part is inside a function to create PVC. I don't see any other instance getting here unless you are really really talented.

**Comment by tuan-hoang1** [comment]
With this patch it stills happens for test `tests/e2e/scale/test_pv_scale_and_respin_ceph_pods.py`

```
================================================================================================================== FAILURES ==================================================================================================================
_______________________________________________________________________ TestPVSTOcsCreatePVCsAndRespinCephPods.test_pv_scale_out_create_pvcs_and_respin_ceph_pods[mgr] _______________________________________________________________________

self = <test_pv_scale_and_respin_ceph_pods.TestPVSTOcsCreatePVCsAndRespinCephPods object at 0x3ff85ac1b50>, namespace = None, storageclass = None, setup_fixture = None, resource_to_delete = 'mgr', memory_leak_function = None

    def test_pv_scale_out_create_pvcs_and_respin_ceph_pods(
        self,
        namespace,
        storageclass,
        setup_fixture,
        resource_to_delete,
        memory_leak_function,
    ):
        pvc_count_each_itr = 10
        scale_pod_count = 120
        size = "10Gi"
        test_run_time = 180
        self.all_pvc_obj, self.all_pod_obj = ([] for i in range(2))
    
        # Identify median memory value for each worker node
        median_dict = helpers.get_memory_leak_median_value()
        log.info(f"Median dict values for memory leak {median_dict}")
    
        # First Iteration call to create PVC and POD
>       self.create_pvc_pod(
            self.rbd_sc_obj, self.cephfs_sc_obj, pvc_count_each_itr, size
        )

tests/e2e/scale/test_pv_scale_and_respin_ceph_pods.py:194: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <test_pv_scale_and_respin_ceph_pods.TestPVSTOcsCreatePVCsAndRespinCephPods object at 0x3ff85ac1b50>, rbd_sc_obj = <ocs_ci.ocs.resources.ocs.OCS object at 0x3ff85989e50>
cephfs_sc_obj = <ocs_ci.ocs.resources.ocs.OCS object at 0x3ff85ae7250>, number_of_pvc = 10, size = '10Gi'

    def create_pvc_pod(self, rbd_sc_obj, cephfs_sc_obj, number_of_pvc, size):
        """
        Function to create multiple PVC of different type and bind mount them to pods
    
        Args:
            rbd_sc_obj (obj_dict): rbd storageclass object
            cephfs_sc_obj (obj_dict): cephfs storageclass object
            number_of_pvc (int): pvc count to be created for each types
            size (str): size of each pvc to be created eg: '10Gi'
        """
        log.info(f"Create {number_of_pvc} pvcs and pods")
        cephfs_pvcs = helpers.create_multiple_pvc_parallel(
            cephfs_sc_obj,
            self.namespace,
            number_of_pvc,
            size,
            access_modes=[constants.ACCESS_MODE_RWO, constants.ACCESS_MODE_RWX],
        )
        rbd_pvcs = helpers.create_multiple_pvc_parallel(
            rbd_sc_obj,
            self.namespace,
            number_of_pvc,
            size,
            access_modes=[constants.ACCESS_MODE_RWO, constants.ACCESS_MODE_RWX],
        )
        # Appending all the pvc obj to base case param for cleanup and evaluation
        self.all_pvc_obj.extend(cephfs_pvcs + rbd_pvcs)
    
        # Create pods with above pvc list
        cephfs_pods = helpers.create_pods_parallel(
            cephfs_pvcs, self.namespace, constants.CEPHFS_INTERFACE
        )
        rbd_rwo_pvc, rbd_rwx_pvc = ([] for i in range(2))
        for pvc_obj in rbd_pvcs:
>           if pvc_obj.get_pvc_access_mode == constants.ACCESS_MODE_RWX:
E           AttributeError: 'list' object has no attribute 'get_pvc_access_mode'

tests/e2e/scale/test_pv_scale_and_respin_ceph_pods.py:61: AttributeError

```

**Comment by shyRozen** [comment]
Using existing cluster which OSD is smaller for this test. Rerun with new cluster with 2048g OSD

**Comment by shyRozen** [comment]
Hi @tuan-hoang1 
As you can see both tests executions passed with the latest code changes.
If you need to test it urgent you can do `git fetch upstream pull/6313/head:scale_fix` and checkout to scale_fix branch.
Will try to merge this as fast as I can.

**Comment by tuan-hoang1** [comment]
Hi @shyRozen, is the `https://patch-diff.githubusercontent.com/raw/red-hat-storage/ocs-ci/pull/6313.patch` same as `git fetch upstream pull/6313/head:scale_fix` ? The branch for this commit looks like `wait_for_resource_list_issue` ?

**Comment by shyRozen** [comment]
@tuan-hoang1 oh it brings the diff. My command create a branch with the changes. Whatever works for you. The commit name is correct.

**Review by ramkiperiy — APPROVED** [review]
LGTM

**Review by ebenahar — APPROVED** [review]

**Comment by shyRozen** [comment]
@tuan-hoang1 The PR is merged so you can use master.

**Pull request merged**